File: frontend/src/types/auditLog.ts
export interface TimelineEntry {
  type: 'audit' | 'event' | 'task'
  action?: string
  field_name?: string
  old_value?: string
  new_value?: string
  subject?: string
  status?: string
  result?: string
  user_id?: string
  user_display_name?: string
  created_at?: string
  reference_id?: string
  reference_type?: string
}

File: backend/app/schemas/audit_log.py
from pydantic import BaseModel
from datetime import datetime


class AuditLogOut(BaseModel):
    id: str
    object_type: str
    object_id: str
    field_name: str | None = None
    old_value: str | None = None
    new_value: str | None = None
    action: str
    user_id: str
    created_at: datetime | None = None

    class Config:
        from_attributes = True


class TimelineEntry(BaseModel):
    type: str  # "audit" | "event" | "task"
    action: str | None = None
    field_name: str | None = None
    old_value: str | None = None
    new_value: str | None = None
    subject: str | None = None
    status: str | None = None
    result: str | None = None
    user_id: str | None = None
    user_display_name: str | None = None
    created_at: datetime | None = None
    reference_id: str | None = None
    reference_type: str | None = None